src/datepicker/ngb-calendar.ts

```diff
@@ -0,0 +1,85 @@
+import {NgbDate} from './ngb-date';
+
+export interface NgbCalendar {
+  getDaysPerWeek(): number;
+  getWeeksPerMonth(): number;
+  getFirstDayOfWeek(): number;
+
+  getWeekDay(date: NgbDate): number;
+  getWeekNumber(week: NgbDate[]): number;
+
+  getWeekDayName(date: NgbDate): string;
+  getMonthName(date: NgbDate): string;
+
+  getNextDate(date: NgbDate): NgbDate;
+  getPrevDate(date: NgbDate): NgbDate;
+
+  getNextMonth(date: NgbDate): NgbDate;
+  getPrevMonth(date: NgbDate): NgbDate;
+
+  getToday(): NgbDate;
+}
+
+export class GregorianCalendar implements NgbCalendar {
```

Comment:
As discussed let's split into 2 services: pure calendar logic and localization-focused part.